### ISSUE TITLE
online autodocs missing

### ISSUE THREAD

**Issue by pazz** [pazz/alot#1324]
Autodoc doesn't seem to work on readthedocs servers again.
https://alot.readthedocs.io/en/latest/api/utils.html

In fact, now that we build the docs while testing anyway, we might just as well upload the latest builds from travis onto a github page.

**Comment by lucc** [comment]
We could also use the `docs` folder and check (on travis like we do already) that the programmer did update all generated files before commiting. The only difference could be that we generate html as well.

**Comment by lucc** [comment]
A first test is here but it does not work right just yet: http://lucc.github.io/alot/index.html

**Comment by lucc** [comment]
But the readme of travis-sphinx looks simpler and clearer than the one of sphinx-deployment. Just my first impression.

**Comment by dcbaker** [comment]
I haven't looked at either, but can we used either to avoid checking generated documents into the git repo and always generate them?

**Comment by pazz** [comment]
@lucc: it looks ok safe for the missing css stuff doesn't it? Which one of the two did you use?

**Comment by lucc** [comment]
No external tools are used, just the generated docs commited to the master branch: https://github.com/lucc/alot/

**Comment by pazz** [comment]
I tried my luck with travis-sphinx but this [doesn't seem to push](https://github.com/Syntaf/travis-sphinx/issues/45). Also, for some magical reason [the tests now fail](https://travis-ci.org/pazz/alot/jobs/452591986). I'm puzzled.

**Comment by pazz** [comment]
diggin around a bit, it seems that travis-sphinx itself uses https://github.com/davisp/ghp-import internally. We might as well call that from travis, as 

```
ghp-import -p -f -r https://$GH_TOKEN@github.com/pazz/alot.git docs/build/html
```

**Comment by pazz** [comment]
I believe the CSS issue is separate, @lucc . I've upluaded the docs manually (via ghp-import) to the gh-pages branch, and github does not seem to pick up on it. Somehow, github does not serve the "_static" directory. You get a 404 for instance on http://pazz.github.io/alot/_static/classic.css
It seems to work for [this project](https://github.com/NASA-DEVELOP/dnppy/tree/gh-pages) though, and I cannot see the difference to our setup.

**Comment by pazz** [comment]
OK, I'm a bit further down the road. Sorry that I had to mess up the commit history on the master branch %)
The docs now build with travis-sphinx and are uploaded as expected to github-pages.
Magically, the css files are now served by github, so it looks OK.
One issue that's left is that we have several travis jobs for "docs", for several python versions, and there seems to be a race among them at the point where they write to the gh-pages branch.

The easiest solution might be to only let one job upload the docs. 
A neater solution would be to generate the docs individually for several versions, and have all of them uploaded. No idea yet how to do this, but I remember having seen sphinx docs with "toggleable" versions..

**Comment by pazz** [comment]
I have fixed the readthedocs build now, it seems like the nicest solution to me as it automatically handles multiple versions etc already. The autodoc problem was that readthedocs used python2 when generating the docs, which lead to syntax errors when importing, hence the missing autodocs.
I have now removed all mentioning of travis-sphinx and polishes the docs somewhat in anticipation of the upcoming release.
